commerce-payment-web/src/main/resources/META-INF/resources/payment_method_action.jsp

```diff
@@ -36,37 +53,121 @@ CommercePaymentMethodGroupRel commercePaymentMethodGroupRel = (CommercePaymentMe
+			<portlet:param name="engineKey" value="<%= engineKey %>" />
 		</portlet:actionURL>
 
 		<liferay-ui:icon
 			message="edit"
 			url="<%= editURL %>"
 		/>
 
-		<portlet:actionURL name="editCommercePaymentMethodGroupRel" var="setActiveURL">
-			<portlet:param name="<%= Constants.CMD %>" value="setActive" />
-			<portlet:param name="redirect" value="<%= currentURL %>" />
-			<portlet:param name="commercePaymentMethodGroupRelId" value="<%= String.valueOf(commercePaymentMethodGroupRel.getCommercePaymentMethodGroupRelId()) %>" />
-			<portlet:param name="active" value="<%= String.valueOf(!commercePaymentMethodGroupRel.isActive()) %>" />
-			<portlet:param name="engineKey" value="<%= commercePaymentMethodGroupRel.getEngineKey() %>" />
-		</portlet:actionURL>
-
 		<liferay-ui:icon
+			id='<%= "togglePaymentMethod-" + modalId %>'
 			message='<%= commercePaymentMethodGroupRel.isActive() ? "deactivate" : "activate" %>'
-			url="<%= setActiveURL %>"
+			url="<%= commercePaymentMethodGroupRel.isActive() ? StringPool.POUND : setActiveURL %>"
 		/>
 
 		<portlet:actionURL name="editCommercePaymentMethodGroupRel" var="editURL">
 			<portlet:param name="<%= Constants.CMD %>" value="<%= Constants.DELETE %>" />
 			<portlet:param name="redirect" value="<%= currentURL %>" />
 			<portlet:param name="commercePaymentMethodGroupRelId" value="<%= String.valueOf(commercePaymentMethodGroupRel.getCommercePaymentMethodGroupRelId()) %>" />
-			<portlet:param name="engineKey" value="<%= commercePaymentMethodGroupRel.getEngineKey() %>" />
+			<portlet:param name="engineKey" value="<%= engineKey %>" />
 		</portlet:actionURL>
 
 		<liferay-ui:icon
 			message="delete"
 			url="<%= editURL %>"
 		/>
 	</c:if>
-</liferay-ui:icon-menu>
\ No newline at end of file
+</liferay-ui:icon-menu>
+
+<c:choose>
+	<c:when test="<%= ((subscriptionEntryCount > 0) || isLastRecurringPaymentMethod) && commercePaymentMethodGroupRel.isActive() %>">
+		<style>
+			.warning-modal .modal-header {
+				background-color: #feefef;
+				border-color: #f48989;
+				color: #da1414;
+			}
+
+			.warning-modal p {
+				font-size: 16px;
+			}
+		</style>
+
+		<div class="warning-modal" id="<%= modalId %>"></div>
+
+		<%
+			String deactivationWarning = StringPool.BLANK;
+
+			if (subscriptionEntryCount > 0) {
+				deactivationWarning += LanguageUtil.format(locale, "there-are-x-subscriptions-that-depend-on-this-payment-method-if-you-deactivate-it-you-will-have-to-manually-manage-those-subscriptions", subscriptionEntryCount);
+			}
+
+			if (isLastRecurringPaymentMethod) {
+				deactivationWarning += "<br /><br />" + LanguageUtil.format(locale, "deactivating-this-payment-method-will-disable-x-subcription-products", recurringCPDefinitionsCount);
+			}
+		%>
+
+		<aui:script use="aui-base">
+			A.use(
+				'aui-modal',
+				function(A) {
+					var <%= modalId %> = new A.Modal(
+						{
+							bodyContent: '<p><liferay-ui:message key="<%= deactivationWarning %>" /><p>',
```

Comment:
we've already localized the warning by this point, so we should remove the `liferay-ui:message` and just put the warning in the `<p>`